### PR TITLE
fix conflict hostPort e2e test

### PR DESCRIPTION
**Pull request by JornShen**
#### What type of PR is this?

/kind bug
/kind flake
/kind failing-test

#### What this PR does / why we need it:

fix network test: [sig-network] HostPort validates that there is no conflict between pods with same hostPort but different hostIP and protocol [LinuxOnly] [Conformance]

failed link: 
https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-e2e-kind-ipv6
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/99098/pull-kubernetes-e2e-kind-ipv6/1361354037437403136/

From the logs of kubelet. pod create no success. The test go through it checking pod no pending.
I think it is no enough to support the following test check.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/network/hostport.go [hunk 1]
```diff
@@ -146,6 +146,7 @@ var _ = SIGDescribe("HostPort", func() {
 				continue
 			}
 			success = true
+			break
```
Comment (aojea) [inline test/e2e/network/hostport.go:149]: I don't know what I was thinking when I did this :smile: , thanks for fixing it

an alternative can be to not use the `success` variable and return directly here, and fail if we go out of the loop
up to you

**Comment by aojea** [comment]
/retest
/lgtm
Thank you very much
/assign @BenTheElder @spiffxp 
this affects conformance

**Comment by JornShen** [comment]
updated. @aojea 
recc

File: test/e2e/network/hostport.go [hunk 2]
```diff
@@ -190,9 +187,11 @@ func createHostPortPodOnNode(f *framework.Framework, podName, ns, hostIP string,
 			NodeName: nodeName,
 		},
 	}
-	_, err := f.ClientSet.CoreV1().Pods(ns).Create(context.TODO(), hostPortPod, metav1.CreateOptions{})
-	framework.ExpectNoError(err)
+	if _, err := f.ClientSet.CoreV1().Pods(ns).Create(context.TODO(), hostPortPod, metav1.CreateOptions{}); err != nil {
+		framework.Failf("error to create pod %s, err:%v", podName, err)
```
Comment (BenTheElder) [inline test/e2e/network/hostport.go:191]: minor grammar nit
```suggestion
		framework.Failf("error creating pod %s, err:%v", podName, err)
```

**Comment by BenTheElder** [comment]
@aojea in what way? seems to me this is functionally from a conformance POV is equivilant?

**Comment by JornShen** [comment]
updated. 
@BenTheElder

**Comment by aojea** [comment]
> @aojea in what way? seems to me this is functionally from a conformance POV is equivilant?

Bad wording, just wanted to say that touches a conformance test to give the pr my priority, not meaning it changes it behaviour sorry
/Lgtm

**Pull request merged**